R/content.R

```diff
@@ -274,6 +274,56 @@ Content <- R6::R6Class(
       cat("\n")
       invisible(self)
     }
+  ),
+  active = list(
+    #' @field default_variant The default variant for this object.
+    default_variant = function(value) {
+      if (missing(value)) {
+        get_variant(self, "default")
+      } else {
+        stop("Default variant cannot be set.", call. = FALSE)
+      }
+    },
```

Comment:
Why does this function accept `value` if it expects `value` to always be "missing"?  Should it be...
```suggestion
    default_variant = function() {
        get_variant(self, "default")
    },
```

Reply:
This is an idiosyncrasy of the active properties in `R6`. Basically, the "value missing" case is a getter, the "value present" case is a setter where `value` is the provided value.

The way this looks in R would be:

```r
object$property <- "value"
```

I didn't want to support changing the default variant on the server by modifying an R property.

Reply:
Ah, I see. Thanks for the clarification.

Reply:
When I've done active bindings in R6 classes before that were not settable, I've done `function()` and it's been fine. What happens if you simplify like @tdstein proposed, does something break?

Reply:
Oop yeah, I misread the documentation — you can just omit the if/else to get the err-on-assignment behavior. Thanks!